graphics/macgui/macwindow.cpp

```diff
@@ -268,9 +269,25 @@ void MacWindow::drawBorderFromSurface(ManagedSurface *g, uint32 flags) {
 }
 
 void MacWindow::setTitle(const Common::String &title) {
-	_title = title;
-	_borderIsDirty = true;
-	_macBorder.setTitle(title, _borderSurface.w, _wm);
+	if (_isTitleVisible) {
+		_title = title;
+		_borderIsDirty = true;
+		_macBorder.setTitle(title, _borderSurface.w, _wm);
+	} else {
+		_shadowedTitle = title;
+	}
+}
+
+void MacWindow::setTitleVisibility(bool visible) {
```

Comment:
I'd rename this method to MacWindow::setTitleVisible and make Director's Window::setTitleVisible override it.

Also move the definitions for `_titleVisible` and `isTitleVisible` on MacWindow instead of Window to eliminate the need for the redundant `_isTitleVisible`.